app/services/ApplicationService.scala

```diff
@@ -220,7 +220,46 @@ class ApplicationService @Inject()(sandboxApplicationConnector: SandboxApplicati
   }
 
   def subscribeToApi(application: Application, context: String, version: String)(implicit hc: HeaderCarrier): Future[ApplicationUpdateResult] = {
-    applicationConnectorFor(application).subscribeToApi(application.id.toString, APIIdentifier(context, version))
+    val applicationConnector: ApplicationConnector = applicationConnectorFor(application)
+
+    val apiIdentifier = APIIdentifier(context, version)
+
+    trait HasSucceeded
+    object HasSucceeded extends HasSucceeded
+
+    def createEmptyFieldValues(fieldDefinitions: Seq[SubscriptionFieldDefinition]) = {
+      fieldDefinitions
+        .map(d => (d.name, ""))
+        .toMap
+    }
+
+    def ensureEmptyValuesWhenNoneExists(fieldDefinitions: Seq[SubscriptionFieldDefinition]): Future[HasSucceeded] = {
+      subscriptionFieldsService.fetchFieldsValues(application, fieldDefinitions, apiIdentifier)
+        .flatMap(values => {
+          if (!values.exists(field => field.value != "")) {
+            val x = subscriptionFieldsService.saveFieldValues(application, context, version, createEmptyFieldValues(fieldDefinitions))
```

Comment:
Remove this temporary variable [x] (I know it was temporary - I introduced it :-( )